dts/bindings/sensor/st,lsm6dso-common.yaml

```diff
@@ -46,6 +46,16 @@ properties:
       mandatory and if not present it defaults to 1 which is the
       configuration at power-up.
 
+  int-open-drain:
+    type: boolean
+    description: |
+      Configure the interrupt pin as open-drain instead of push-pull.
+
+  int-active-low:
+    type: boolean
+    description: |
+      Configure the interrupt pin as active low instead of active high.
+
```

Comment:
This should be encoded already in `irq-gpios`. In devicetree's phandle array you should set that property to something like `irq-gpios = <&gpio0 12 (GPIO_ACTIVE_LOW | GPIO_OPEN_DRAIN)>;`

Reply:
Great points. It wasn't clear to me if `GPIO_OPEN_DRAIN` applied to input pins or not, as that's what `irq-gpios` is controlling. These two settings I have added control the interrupt output pin of the device. We can certainly do the active high/active low by reading flags from `irq-gpios`.

Looking again at the lis2dh driver, we see a different strategy - there is an integer field for the type of interrupt (both edges, rising, falling, high, low). This could also have been encoded in the `irq-gpios` field, but wasn't.

I'm open to your approach, but I'd be keen to see some precedence.

Reply:
@MaureenHelm should be looped into the discussion